app/models/user.rb

```diff
@@ -47,6 +49,10 @@ def set_reset_password_token
     raw
   end
 
+  def set_tour_steps
+    WelcomeTour.new.set_user_steps self
```

Comment:
This is more easy to see that you are initializing a user attributes:
```ruby
def init_tour_steps
  self.tour_steps = WelcomeTour::STEPS.to_json
end
```
and you can remove the method `set_user_steps` from WelcomeTour.